source/server/async_operation.h

```diff
@@ -0,0 +1,68 @@
+#ifndef NIDEVICE_GRPC_DEVICE_ASYNC_OPERATION_H
+#define NIDEVICE_GRPC_DEVICE_ASYNC_OPERATION_H
+
+#include <grpcpp/grpcpp.h>
+
+#include <memory>
+
+#include "async_method_context.h"
+#include "async_processing_loop.h"
+
+using nidevice_grpc::SharedMethodContextPtr;
+
+namespace nidevice_grpc {
+// Calls writer.Finish and then releases the method_context_.
+template <typename TRequest, typename TResponse>
```

Comment:
Note: these are templated because they own a templated `SharedMethodContextPtr<TRequest, TResponse>`. I can fix that with some type-erasing/base-class, which would allow moving these implementations out of the header. I (probably) plan to do that, but want to make sure I'm on the right track first.